R/pkg/R/sparkR.R

```diff
@@ -344,6 +344,7 @@ sparkRHive.init <- function(jsc = NULL) {
 #' @note sparkR.session since 2.0.0
 sparkR.session <- function(
   master = "",
+  deployMode = "",
```

Comment:
I don't think we should add the parameter in the middle which would break user passing parameter by order.

On the other hand, do we need this as a formal parameter? Most cluster setup I know of is already having deploy mode in spark conf - and to @shivaram 's point, why isn't SPARK_HOME set? If it is, we wouldn't be downloading anyway, right?